docs/notes/2.24.x.md

```diff
@@ -90,6 +90,10 @@ Mypy will now typecheck previously-ignored python sources without a `.py` or `.p
 
 The `pants.backend.url_handlers.s3` backend now correctly passes along query parameters such as `versionId` for `s3://` urls.
 
+#### Semgrep
+
+The default version of [the semgrep tool](https://www.pantsbuild.org/2.24/reference/subsystems/semgrep) has been upgraded from 1.86.0 to 1.94.0.
```

Comment:
```suggestion
The default version of [the semgrep tool](https://www.pantsbuild.org/2.24/reference/subsystems/semgrep) has been upgraded from 1.86.0 to [1.94.0](https://github.com/semgrep/semgrep/releases/tag/v1.94.0).
```

Reply:
Sounds good, I'll add it. 

Idle thinking: I'm not sure what to do about 1.87-1.93, linking to each one individually seems like overkill/too much work, but they're all relevant to the upgrade. 🤷